src/mo_idl/idl_to_mo.ml

```diff
@@ -97,24 +97,34 @@ and check_meth env occs (m: typ_meth) =
 
 let check_prog (env: typ I.Env.t) actor : M.typ =
   let occs = ref M.Env.empty in
-  match actor with
-  | Some {it=ServT ms; _} ->
-     let fs = List.map (check_meth env occs) ms in
-     (* TODO: why do we only check and include the mentioned types (occs),
-        and not all of the .did declared ones (available to the caller, if not here? *)
-     let fs = M.Env.fold (fun id t fs ->
+  let fs = match actor with
+    | Some {it=ServT ms; _} ->
+      List.map (check_meth env occs) ms
+    | Some {it=ClassT (ts1, t); at; _} ->
+      (*@HACK: import service constructors as instantiated services *)
+      (*TODO: fix dfx to derive the correct instantiated candid instead *)
+      begin
+        let t' = check_typ' env occs t in
+        match M.normalize t' with
+        | M.Obj (M.Actor, fs) ->
+          Diag.print_messages [Diag.warning_message at "M0163" "import"
```

Comment:
Maybe obsolete the old error code and create a new one? Aren't they designed to be immutable?

Reply:
Ok.

Reply:
Done